contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -323,9 +326,47 @@ contract TrueCreditAgency is UpgradeableClaimable {
     }
 
     function poke(ITrueFiPool2 pool) public {
+        uint256 bitMap = usedBucketsBitmap;
+        uint256 timeNow = block.timestamp;
+
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
+                continue;
+            }
+
+            _pokeSingleBucket(pool, uint8(i), timeNow);
+        }
+    }
+
+    function pokeSingleBucket(ITrueFiPool2 pool, uint8 bucketNumber) internal {
```

Comment:
Do we need `pokeSingleBucket()`? Can't we just use `_pokeSingleBucket()`?

Reply:
Sometimes timeNow has to be calculated outside of single bucket poke, but more often we can integrate it inside the funciton